armonik-client/src/main/java/fr/aneo/armonik/client/model/TaskSubmitter.java

```diff
@@ -269,23 +291,29 @@ private static CompletionStage<List<BlobId>> getIdsFrom(Collection<BlobHandle> b
   private IntFunction<BlobHandle> toBlobHandle(SessionId sessionId, CompletionStage<List<ResultRaw>> resultRaws) {
     return index -> new BlobHandle(
       sessionId,
-      resultRaws.thenApply(resultRaw -> new BlobInfo(BlobId.from(resultRaw.get(index).getResultId()))),
+      resultRaws.thenApply(resultRaw -> new BlobInfo(
+        BlobId.from(resultRaw.get(index).getResultId()),
+        sessionId,
+        resultRaw.get(index).getName(),
+        resultRaw.get(index).getManualDeletion(),
```

Comment:
can you get it only once in a multi statement lambda

Reply:
I have extracted the creation of the BlobInfo in another method so it `toBlobHandle()` method clearer.